src/CppAst/CppModelBuilder.cs

```diff
@@ -85,6 +86,39 @@ private CppType TryToCreateTemplateParameters(CXCursor cursor, void* data)
             return null;
         }
 
+
+        /*
+         * Helper function to add specialized template parameters from a given source to a destination parameter list
+         * It also takes care of the fact that libClang can return the same template parameter 'a bit differently' 
+         * depending on where it is being used
+         */
+        private void AddTemplateSpecializedArguments(List<CppTemplateArgument> sourceArguments, List<CppTemplateArgument> destinationArguments)
+        {
+            foreach (var p in sourceArguments)
+            {
+                if (!destinationArguments.Contains(p))
+                {
+                    /** 
+                     * Ok, BaseTypes is a strange beast. the same template parameter
+                     * might be received "a bit differently" from the bases, so we do 
+                     * a 'simpler compare' as well, e.g. check if the source param
+                     * is the same.
+                     * Please note that the "ideal" check would be to make sure that if
+                     * we are setting the same template parameter multiple times, we are
+                     * setting that to the same value. HOWEVER this should not happen
+                     * as it would mean we are adding specialized template parameters 
+                     * from different consumers into the very same base type -> coding error.
+                     */
+                    var numExistingParams = destinationArguments.Where(param => param.SourceParam.FullName.Equals(p.SourceParam.FullName)).ToList().Count;
```

Comment:
I am still a little bit confused about this check. Are both checks (first check if the parameter itself exist in `destinationArguments`, then check if a parameter with the *same name* exists in `destinationArguments`) needed? I feel like that one of the checks is not needed, in this case it seems to be the if statement.

This can also be simplified to:
```suggestion
                    bool isExistingParams = destinationArguments.Where(param => param.SourceParam.FullName.Equals(p.SourceParam.FullName)).ToList().Any();
```

Reply:
Rephrased the comment, rewritten the check